jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -504,13 +506,15 @@ public void makeTempDirectory(File parent, WebAppContext context)
             //if it is to be persisted, make sure it will be the same name
             //by not using File.createTempFile, which appends random digits
             tmpDir = new File(parent, temp);
+            configureTempDirectory(tmpDir, context);
         }
         else
         {
-            //ensure file will always be unique by appending random digits
+            // ensure dir will always be unique by having classlib generate random path name
             tmpDir = Files.createTempDirectory(parent.toPath(), temp).toFile();
+            tmpDir.deleteOnExit();
+            ensureTempDirUsable(tmpDir);
```

Comment:
This execution path does not use `configureTempDirectory`.
It's always a new / and unique directory.
And it's always `!isPersisTempDirectory()`